### ISSUE TITLE
No MIDI Note output to VCV Rack VST

### ISSUE THREAD

**Issue by Teteros** [kmatheussen/radium#1160]
### 1. Steps to reproduce the problem.

* Step 1: Download and run [VCV Rack](https://vcvrack.com). 0.6
* Step 2: Open Radium, load the VST Bridge (`Bridge/VCV-Bridge.so`) as a new **audio** instrument and connect its playback to System Output.
* Step 3: Start VCV Rack, right click on the canvas and add "MIDI-1" and "Audio" (or Scope) modules from Core.
* Step 4: Set them to `Bridge` mode and match the Port with the VST Port used in Radium (I left it at 1)
* Step 5: Connect `CV` / `Gate` or `Vel` from MIDI-1 to `1` or `2` on Audio (or preferably Scope from Fundamental if available)
* Step 6: Play a note in Radium

* I expected this to happen: Sound (V) from any of the outputs of MIDI-1

* This is what actually happened: Only CLK1 and CLK2 output pulses, which are coming from Rack itself, Radium is silent on all events.
Right clicking the VST in mixer => Show Info reveals radium thinks VCV Bridge `Accepts MIDI: No`

This is only for Note MIDI events, MIDI-CC module in Rack works fine, meaning automation of CC 0-15 seems to works from radium -> rack.

Rack Docs on these modules:
https://vcvrack.com/manual/Core.html#midi-1
https://vcvrack.com/manual/Bridge.html

### 2. Which version of Radium are you using? Do you know if an earlier version has worked?
5.9.10, VCV Rack bridge [README](https://github.com/VCVRack/Bridge#linux) hints radium 5.4.9 may have worked before/was tested.

### 3. If relevant, which operating system did you run Radium on?
LXVST on Arch Linux x64 (radium/rack compiled from source)

**Comment by GeorgeNs** [comment]
Yes, this is a known issue due to buffer setting. That's sad because rack is awesome open source piece and its plugin manager is so awesome too. Ardour works fine with bridge though.
I wish a big radium update for the version 6.-)

**Comment by kmatheussen** [comment]
I've tried the recipe quickly but didn't understand "match the Port with the VST Port used in Radium (I left it at 1)". I skipped this step, and tried next step, but didn't understand that one either. Is this a bug in Radium? Should I invest time on this ticket?

**Comment by Teteros** [comment]
Thanks for looking into this, I've made a screencast to demonstrate the issue:

[![screencast_link](https://user-images.githubusercontent.com/5067989/51096866-b97de380-17b7-11e9-90ab-eb12082a638a.png)](https://u.teknik.io/FlDGk.webm)

> Is this a bug in Radium? 

I think it is because radium reports the plugin does not accept MIDI which isn't true.
![infobox](https://user-images.githubusercontent.com/5067989/51097028-ebdc1080-17b8-11e9-911c-60bca7e0b78c.png)

Other synths that do play notes in the sequencer report `Yes`.

The VCVRack bridge readme mentions trouble if the VST host "does not use an audio buffer with a fixed size" but the issue is that radium does not seem to send the note events to the VST bridge at all.

**Comment by kmatheussen** [comment]
Thank you very much. I see the problem now. I've digged a little bit, and things only get more confusing. First of all, it seems like the vst plugin doesn't report immediately that it is able to receive MIDI. But after taking that into account, it still won't receive MIDI, at least not from the editor. I am, for some reason, able to send MIDI when using the virtual keyboard (the one Radium adds under the native VST gui), which makes no sense.

**Comment by kmatheussen** [comment]
Oh, and radium should use a fixed size for the audio buffer.

**Comment by kmatheussen** [comment]
By the way. When Radium reports that the plugin doesn't receive MIDI, it still tries to send MIDI to the plugin when it receives notes. So that's just a display issue. I don't know what's happening here though. Has there been reports of the same problem for hosts that doesn't use fixed buffers?

**Comment by Teteros** [comment]
I've tested Carla 1.9.13 (2.0-RC3) now and the "fixed-size buffer" option on/off makes no difference at least.

Hm, I would imagine the audio buffers would only be relevant when using the audio input/output module (You can set that to `Bridge` mode rather than ALSA/JACK/etc to route audio from/to DAW)

It's weird that only MIDI-1 module is misbehaving. MIDI-CC used to transmit automation works well in radium.

**Comment by kmatheussen** [comment]
> It's weird that only MIDI-1 module is misbehaving. MIDI-CC used to transmit automation works well in radium.

I agree. That is really strange. And even stranger is notes from the virtual keyboard works, but not notes from the editor. Both sends events to the same buffer. Since this doesn't work in Traction either, it's probably something in JUCE causing this since both programs use JUCE to host VST plugins.

**Comment by Teteros** [comment]
For the record the audio input/output on bridge mode works too.
Passes sound from JACK as expected.
![radium_vcvrack_ports](https://user-images.githubusercontent.com/5067989/51103119-82b8c500-17d9-11e9-9078-0c4501bc36df.png)

It's just the `MIDI-1` that seems bugged for the time being, yes.
